src/github/github.py

```diff
@@ -0,0 +1,188 @@
+
+    return result
+
+async def _github_request_post(
+        url: str,
+        token: str,
+        post_data: Any
+) -> Response:
+    result = requests.post(
+        url,
+        headers={
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {token}",
+            "X-GitHub-Api-Version": "2022-11-28"
+        },
+        data=post_data
+    )
+    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
+    if rate_limit_remaining < 50:
+        raise Exception("Less than 50 api calls remaining before being rate limited, please try again later")
+
+    return result
+
+async def _github_request_delete(
+        url: str,
+        token: str
+) -> Response:
+    result = requests.delete(
+        url,
+        headers={
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {token}",
+            "X-GitHub-Api-Version": "2022-11-28"}
+    )
+    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
+    if rate_limit_remaining < 50:
+        raise Exception("Less than 50 api calls remaining before being rate limited, please try again later")
+
+    return result
+
+async def _github_request_put(
+        url: str,
+        token: str,
+        put_data: Any
+) -> Response:
+    result = requests.put(
+        url,
+        headers={
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {token}",
+            "X-GitHub-Api-Version": "2022-11-28"},
+        data=put_data
+    )
+    rate_limit_remaining = int(result.headers["x-ratelimit-remaining"])
+    if rate_limit_remaining < 50:
+        raise Exception("Less than 50 api calls remaining before being rate limited, please try again later")
+
+    return result
+
+async def get_user_by_username(
+        username: str
+) -> GithubUser:
+    """
+        Takes in a Github username and returns an instance of GithubUser.
+
+        May return an empty list if no such user was found.
+    """
+    result = await _github_request_get(f"https://api.github.com/users/{username}",
+                              os.environ.get("GITHUB_TOKEN"))
+    result_json = result.json()
+    return [GithubUser(user["login"], user["id"], user["name"]) for user in [result_json]]
+
+async def get_user_by_id(
+    uid: str
+) -> GithubUser:
+    """
+        Takes in a Github user id and returns an instance of GithubUser.
+
+        May return an empty list if no such user was found.
+    """
+    result = await _github_request_get(f"https://api.github.com/user/{uid}",
+                              os.environ.get("GITHUB_TOKEN"))
+    result_json = result.json()
+    return [GithubUser(user["login"], user["id"], user["name"]) for user in [result_json]]
+
+async def add_user_to_org(
+        org: str = github_org_name,
+        uid: str | None = None,
+        email: str | None = None
```

Comment:
I would appreciate a quick docstring saying that at least one of these is required to call this function

Reply:
Resolved in be0659a

Reply:
this is good, but fyi docstrings ```"""this is a docstring"""``` as the first line of a function is picked up by the intellisense.

Reply:
Ah you're right. 29c04e2